core/src/main/scala/org/apache/spark/deploy/worker/ExecutorRunner.scala

```diff
@@ -185,11 +185,11 @@ private[deploy] class ExecutorRunner(
 
       // Redirect its stdout and stderr to files
       val stdout = new File(executorDir, "stdout")
-      stdoutAppender = FileAppender(process.getInputStream, stdout, conf)
+      stdoutAppender = FileAppender(process.getInputStream, stdout, conf, true)
 
       val stderr = new File(executorDir, "stderr")
       Files.write(header, stderr, StandardCharsets.UTF_8)
-      stderrAppender = FileAppender(process.getErrorStream, stderr, conf)
+      stderrAppender = FileAppender(process.getErrorStream, stderr, conf, true)
```

Comment:
what about we wait both threads to finish first by `stderrAppender.awaitTermination()`, close the input streams, and call `Process.waitFor`?

Reply:
or I think you can add one more method in `FileAppender` that sets a bool variable (like `markedForStop` and `stop`).

Reply:
That will make the changes minimised.

Reply:
Would waiting for termination possibly make it hang? same with a boolean flag. I think it maybe needs forcibly closing

Reply:
Yeah, if we're worried, that's fine on that approach.

> I think you can add one more method in FileAppender that sets a bool variable (like markedForStop and stop).

what about this though? this will minimize the changes

Reply:
Would that make the invoking complicate? 
On the other hand, if the constructor and "set bool variable" in two statements, there is still a chance that the inputStream would not been closed: the consume thread is started in construtor, if the thread exited immediately due some error, the inputStream may not close since the "set bool variable" may not called at that time.

Reply:
Hm, okie. I am fine